session/pingpong/accountant_promise_settler.go

```diff
@@ -248,23 +255,54 @@ func (aps *AccountantPromiseSettler) listenForSettlementRequests() {
 		case <-aps.stop:
 			return
 		case p := <-aps.settleQueue:
-			aps.settle(p)
+			go aps.settle(p)
 		}
 	}
 }
 
-func (aps *AccountantPromiseSettler) settle(p receivedPromise) {
+// ErrNothingToSettle indicates that there is nothing to settle.
+var ErrNothingToSettle = errors.New("nothing to settle for the given provider")
+
+// ForceSettle forces the settlement for a provider
+func (aps *AccountantPromiseSettler) ForceSettle(providerID, accountantID identity.Identity) error {
+	promise, err := aps.promiseStorage.Get(providerID, accountantID)
+	if err == ErrNotFound {
+		return ErrNothingToSettle
+	}
+	if err != nil {
+		return errors.Wrap(err, "could not get promise from storage")
+	}
+
+	hexR, err := hex.DecodeString(promise.R)
+	if err != nil {
+		return errors.Wrap(err, "could not decode R")
+	}
+
+	promise.Promise.R = hexR
+	return aps.settle(receivedPromise{
+		promise:  promise.Promise,
+		provider: providerID,
+	})
+}
+
+// ErrSettleTimeout indicates that the settlement has timed out
+var ErrSettleTimeout = errors.New("settle timeout")
+
+func (aps *AccountantPromiseSettler) settle(p receivedPromise) error {
 	aps.setSettling(p.provider, true)
 	log.Info().Msgf("Marked provider %v as requesting setlement", p.provider)
 	sink, cancel, err := aps.bc.SubscribeToPromiseSettledEvent(p.provider.ToCommonAddress(), aps.config.AccountantAddress)
 	if err != nil {
 		aps.setSettling(p.provider, false)
 		log.Error().Err(err).Msg("Could not subscribe to promise settlement")
-		return
+		return err
 	}
+
+	ch := make(chan error)
```

Comment:
Could be a bit more clear name for ch. Maybe name it errc which means error channel.